ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java

```diff
@@ -218,15 +214,32 @@ private void refreshIngestionAuthToken() throws IngestionClientException, Ingest
                     identityToken = resultTable.getString(0);
                 }
             } catch (DataServiceException e) {
-                throw new IngestionServiceException(e.getIngestionSource(), "Error in refreshing IngestionAuthToken", e);
+                throw new IngestionServiceException(e.getIngestionSource(), "Error refreshing IngestionAuthToken", e);
             } catch (DataClientException e) {
-                throw new IngestionClientException(e.getIngestionSource(), "Error in refreshing IngestionAuthToken", e);
+                throw new IngestionClientException(e.getIngestionSource(), "Error refreshing IngestionAuthToken", e);
             } finally {
                 authTokenLock.writeLock().unlock();
             }
         }
     }
 
+    protected String getServiceType() throws IngestionServiceException, IngestionClientException {
+        log.info("Getting version to determine endpoint's ServiceType");
+        try {
+            KustoOperationResult versionResult = client.execute(Commands.VERSION_SHOW_COMMAND);
+            if (versionResult != null && versionResult.hasNext() && versionResult.getResultTables().size() > 0) {
+                KustoResultSetTable resultTable = versionResult.next();
+                resultTable.next();
+                return resultTable.getString(SERVICE_TYPE_COLUMN_NAME);
+            }
+        } catch (DataServiceException e) {
+            throw new IngestionServiceException(e.getIngestionSource(), "Error getting version", e);
```

Comment:
Method name is getServiceType yet all exceptions mentions the inner implementation of using ".show version" command. 
Consider export the command execution to a different method and handle it's execution there.

Reply:
Here's my thought process:
- The method should be named according to what it does. Since it gets the service type, it's called getServiceType()
- The meat of this is 5 lines of code, 1 of which executes ".show version" and 4 of which get the ServiceType. So separating this into 2 methods seems to be an antipattern (https://mytwoocents.com/2017/03/25/one-line-methods-can-easily-become-an-antipattern/)
- The exceptions describe what went wrong, which in this case can only be the ".show version" since that's an external call. The getting of the ServiceType just doesn't throw any exceptions.

I like what you're saying, though. I'm going to add to the exception message to include that ultimately we're trying to get the ServiceType. Let me know if that fixes it.

Reply:
I think the method name is good because its describes **what** it does, the exception in the other hand tells me **how** it does it. I don't think its too bad to have another method but if you like you can just change the message.